packages/cw4/src/helpers.rs

```diff
@@ -79,6 +80,17 @@ impl Cw4Contract {
         Map::new(MEMBERS_KEY).query(querier, self.addr(), addr)
     }
 
+    /// Check if this address is a member, and if its weight is >= 1
+    pub fn is_voting_member<T: Into<String>>(
```

Comment:
This is now checking for membership, not voting rights.

Why not keeping the impl as before (two checks, one for membership, the other for weight >= 1) and use `is_member` where only a membership check is needed.

Moreover: the signatures of both functions are not consistent. `is_member` is returning an `Option`, whereas `is_voting_member` is returning a `Result`. And `is_member` is not taking a height parameter.

What I would suggest is:

- Create a new `member_weight` function, that works like the current `is_member` (Besides, it takes an `Option<height>` param for generality, and so allows for snapshotted and not snapshotted queries).
- Make both `is_*` functions work the same. They can both receive an `Option<height>` and allow for both query types (snapshotted or not). They both then return an error if their "is" condition is not satisfied.

You can then implement both `is_` functions in terms of `member_weight`, probably.

This is more work, but the interface will be both consistent, clear, and general.

Reply:
Since voting power (weight) is required both in proposing and voting implementations, it made no sense to create separate third helper `member_weight`. 
I unified implementations of `is_member` and `is_voting_member` basically as you've suggested. 
`is_voting_member` will return `Option` with weight, but only if `weight >= 1`.